### DISCUSSION TITLE
Division by pages

### DISCUSSION THREAD

**Discussion by Marion64** [vers-one/EpubReader#73] [Ideas]
Hello, your library is very good for my need, however it's missing a very important feature that would make it perfect. If you could implement dividing the epub content by page and not only by navigation, then that would be amazing.

Thank you

**Comment by vers-one** [comment]
@Marion64, what would be the page size for such division?

**Comment by Marion64** [comment]
@vers-one If it's possible to create a function that accepts input of number of characters (including whitespaces and linebreaks) per page that would be great. Else, I'd say the average of 1800 characters (including whitespaces and linebreaks) per page would be good. If you have a better way of implementing this idea then feel free to share your thoughts.

Thank you

**Comment by vers-one** [comment]
If we split the content into pages just by the number of characters, then such split will be uneven. Consider the following examples:

This page has 2701 characters:
![Example 1](https://user-images.githubusercontent.com/12114169/205931646-95bc9872-ee84-4611-beab-98fc737aa3b5.png)

And this same size page has only 1513 characters:
![Example 2](https://user-images.githubusercontent.com/12114169/205931990-b11c9cca-d966-4f42-b51d-3b8a1a993e63.png)

These are the pages from the same book. Additionally, images will add more inconsistencies into such page splits.

In order to split the content into pages evenly, with respect to the device's screen size, fonts, and scaling preferences, it needs to be rendered first. Then the rendered content needs to be divided vertically into even chunks. All of this can be done only on the application side.

**Reply by Marion64** [reply]
@vers-one I'm sorry for my long absence, I was very sick the last couple of days. I appreciate your quick response.

Just to let you know what I'm trying to do, I'm developing a WPF software for converting text to speech for sighted users. The goal is to make it as easy and accessible as possible.

You have a point about the division balance issue, thank you for explaining. How about dividing by word count, would that solve the issue? I think the average word count for books is 300 per page. Words such as, stairs!" or nose-- would count as 1 word still, if there are lone characters such as ! or 2 specials characters such as ?" then they would count as a word. 

Let me know what you think.

Thank you

**Reply by vers-one** [reply]
Using word count would still produce an uneven division. And that also doesn't solve the problem with images. Consider the following page from the same book on the same device:
![Example 1](https://user-images.githubusercontent.com/12114169/207497099-c09e0ac5-928b-47cd-ba03-a95a83c309a4.png)

This page contains much fewer words than other pages from the same book that don't have images.

Now imagine that a page gets rendered on a 32" display on a PC or a Mac:
![Example 2](https://user-images.githubusercontent.com/12114169/207498657-5412069b-4f83-469c-88f6-55e8d2ab4cb2.png)

and compare it with a page rendered on a 6" phone:
![Example 3](https://user-images.githubusercontent.com/12114169/207498776-6584b194-bd1b-4b64-a1cd-e017a3c8fd1f.png)

As you can see, the screen size of the device is a huge factor that affects the size of the page.

Most reader apps also let the user choose the size of the font, making it larger or smaller which in turn determines how much content will fit on a page.

EpubReader library cannot make any assumptions about the screen size of the device or the scaling options set by the user which is why this task has to be done on the application side. If your application only needs to convert the text of the book to speech and it doesn't need to render it on a screen, then you can easily divide the textual content into chunks of 300 words, after extracting it from HTML.

**Reply by Marion64** [reply]
@vers-one Yes, I see.. Ok, thank you for your time and effort. I will go ahead and try to come up with a custom solution to the problem I'm facing.

Thank you
